js/builtins/weakrefs/gc-has-one-chance-to-call-cleanupCallback-queueMicrotask.optional.any.js

```diff
@@ -0,0 +1,105 @@
+// META: script=resources/maybe-garbage-collect.js
+// ├──> maybeGarbageCollectAndCleanupAsync
+// ├──> maybeGarbageCollectAsync
+// └──> resolveGarbageCollection
+/*---
+esid: sec-finalization-registry-target
+info: |
+  FinalizationRegistry ( cleanupCallback )
+
+  FinalizationRegistry.prototype.cleanupSome ( [ callback ] )
+
+  ...
+  4. If callback is not undefined and IsCallable(callback) is false, throw a TypeError exception.
+  5. Perform ? CleanupFinalizationRegistry(finalizationRegistry, callback).
+  6. Return undefined.
+
+  Execution
+
+  At any time, if an object obj is not live, an ECMAScript implementation may perform the following steps atomically:
+
+  1. For each WeakRef ref such that ref.[[Target]] is obj,
+    a. Set ref.[[Target]] to empty.
+  2. For each FinalizationRegistry finalizationRegistry such that finalizationRegistry.[[Cells]] contains cell, such that cell.[[Target]] is obj,
+    a. Set cell.[[Target]] to empty.
+    b. Optionally, perform ! HostCleanupFinalizationRegistry(finalizationRegistry).
+---*/
+
+
+let cleanupCallback = 0;
+let called = 0;
+function cb(holding) {
+  assert_equals(holding, 'a');
+  called += 1;
+}
```

Comment:
In WPT, an uncaught exception will be interpreted as a harness error rather than a test failure. I recommend pushing the callback's argument value to an array. Then, you can assert the value in the test body (you could also replace `called` with the length of the new array).

That would also be a good improvement for [the Test262 test upon which this code is based](https://github.com/tc39/test262/blob/5a01caed71610d9484171206f01d15d537c4a5aa/test/built-ins/FinalizationRegistry/gc-has-one-chance-to-call-cleanupCallback.js) (that is, even where there is no error/failure distinction). As written, the test relies on the implementation to correctly propagate abrupt completions originating from the callback to `cleanupSome`. Verifying that specific behavior with a new test will reduce the risk of false positives in this one.